### PR TITLE
Feat/search product

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/76015

## WHAT (optional)
- [Guest] Search and filter products

## HOW
- Implement search and filter products
- Implement paging

## Evidence (Screenshot or Video)
image
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/services/product.service.ts [hunk 1]
```diff
@@ -47,4 +50,54 @@ export class ProductService {
...
     });
   }
+  
+  public async getProductPage(pageOptionsDto: ProductPageOptions) 
+  :Promise<PageDto<ProductEntity>> {
+    const query = this.productRepository.createQueryBuilder('product')
+    .leftJoinAndSelect('product.categories', 'category');
+    
+    // Handle filter
+    if(pageOptionsDto.categoryIds && pageOptionsDto.categoryIds.length > 0) {
```
Comment (huongnt-2545) [inline src/services/product.service.ts]: e đặt biến để tránh gọi dài dòng nha
vd: 
```
const {categoryIds, minPrice, ...} = pageOptionsDto
```

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**